### PR TITLE
Improving AI castle defenses

### PR DESCRIPTION
**Pull request by idshibanov**
Marking threatening enemy armies and friendly castles under attack as priority targets. AI heroes now have incentives to go to the castle to defend it and switch into sleeper mode to stay there as long as threat persists.

### CODE REVIEW

File: src/fheroes2/ai/ai.h [hunk 1]
```diff
@@ -62,6 +62,19 @@ namespace AI
         EXPLORER
     };
 
+    // Although AI heroes are capable to find their own tasks we can strategic AI should be able to focus them on most critical tasks
```
Comment (ihhub) [inline src/fheroes2/ai/ai.h:65]: Some spelling issue in this part of the sentence `find their own tasks we can strategic AI`. I think we should remove `we can` part.

File: src/fheroes2/ai/normal/ai_normal.h [hunk 2]
```diff
@@ -197,6 +198,7 @@ namespace AI
 
         double getObjectValue( const Heroes & hero, const int index, const double valueToIgnore, const uint32_t distanceToObject ) const;
         int getPriorityTarget( const HeroToMove & heroInfo, double & maxPriority );
+        bool isCriticalTask( const int index ) const;
```
Comment (ihhub) [inline src/fheroes2/ai/normal/ai_normal.h:201]: I am suggesting to make this method inline as it literally consists of 1 line of code.

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @idshibanov , I put just few small comments here. Could you please check them?

**Review by ihhub — APPROVED** [review]

**Review by oleg-derevenetz — APPROVED** [review]

**Comment by ihhub** [comment]
@idshibanov , great work! Thanks a lot for these changes!

**Pull request merged**